src/test/java/org/folio/inventory/resources/ItemStatusURLTest.java

```diff
@@ -0,0 +1,30 @@
+package org.folio.inventory.resources;
+
+import org.folio.inventory.domain.items.ItemStatusName;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import junitparams.Parameters;
+import junitparams.JUnitParamsRunner;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(JUnitParamsRunner.class)
+public class ItemStatusURLTest {
+  @Parameters(
+    {
+      "/mark-in-process,In process",
+      "/mark-in-process-non-requestable,In process (non-requestable)",
+      "/mark-intellectual-item,Intellectual item"
+    }
+  )
+  @Test
+  public void canGetItemStatusByFullUrlAndCanGetUrlByStatusName(String url, String statusName) {
+    String dummyUrl = "http://dummy.net/239853$URL$";
+    final var itemStatus= ItemStatusURL.getItemStatusNameForUrl(dummyUrl.replace("$URL$",url));
+    assertThat(itemStatus.isPresent()).isTrue();
+    ItemStatusName itemStatusName = ItemStatusName.forName(statusName);
+    assertThat(itemStatus.get()).isEqualTo(itemStatusName);
+    var itemStatusUrl = ItemStatusURL.getUrlForItemStatusName(itemStatusName);
+    assertThat(itemStatusUrl.isPresent()).isTrue();
+    assertThat(itemStatusUrl.get()).isEqualTo(url);
+  }
```

Comment:
This appears to be two different scenarios:
* Determining the URL for a given target status
* Determining the intended target status for a given URL

I think splitting them up would help the reader to understand them better and provide better diagnostics when the tests fail.

Reply:
May be it have a sense but looking on the other side does it required additional set of parameters which will need additionally manage when workflow will changed. As for me each manually managed block - is a source for mistakes. May be adding the comments would enough?

Reply:
A good start could be to separate these things into sections in the code with comments.

I agree that ideally we wouldn't need to maintain another map of statuses to URLs. I think that could be avoided by changing the production code to use explicit configuration.